### ISSUE TITLE
I'm watching my SCSS with addWatchTarget. How do I trigger the SCSS to actually compile?

### ISSUE THREAD

**Issue by arrowtype** [11ty/eleventy#817]
I'm excited to be trying 11ty for the first time, after hearing great stuff about it for months and enjoying the recent ShopTalkShow with Zach. :)

Getting templates to build has been fairly simple, but I'm now getting stuck on hot-reloading with changes to my SCSS.

## The Problem

I've upgraded to [v0.10.0](https://github.com/11ty/eleventy/releases/tag/v0.10.0-beta.1) so I can use `addWatchTarget()` in the eleventy config. This is working pretty well: when I make a change to my SCSS files, eleventy reloads the site. _However,_ it doesn't actually process the SCSS changes – it's only reloaded the nunjucks template files. 

```js
// .eleventy.js
module.exports = function(eleventyConfig) {
    eleventyConfig.addWatchTarget("./src/styles/");
    return {
        dir: {
            input: "src",
            output: "dist",
            includes: "_includes",
            layouts: "_layouts"
        },
        templateFormats: ['njk', 'md', 'scss', 'css', 'html', 'yml'],
        htmlTemplateEngine: "njk"
    };
  };
```

To actually get the CSS changes into my site, I need to stop the browserSync server, run node-sass to build the CSS, and then restart the server.

## My expectation

My expectation is that with `addWatchTarget`, once I use `eleventy --serve`, Eleventy would start watching and compiling those SCSS files. Or at least, I would hope that it would offer some way for me to cue a script such as `npm run styles:dev` to cue that compilation and inject the new CSS. 

Is there something I'm missing? How are people handling hot reloading for SCSS changes? I'm not married to this particular solution; I'm just hoping for a simple way to use SCSS with the basic browserSync path that the 11ty docs suggest in Getting Started.

Thanks for any insights!

---

Here's my repo at the point of the issue: https://github.com/arrowtype/11ty-test/tree/22b1feb767a7760346bdb5f151ca086b8b51939a

**Comment by arrowtype** [comment]
It seems like perhaps, using `node-sass-chokidar` might be one way to watch SCSS changes, as recommended in this tutorial: https://github.com/hellobrian/sass-recipes/tree/master/node-sass

However, I think using `node-sass-chokidar` would effectively replace the purpose of `addWatchTarget` completely, correct? If so, is there any purpose for `addWatchTarget`, outside of my imagined purpose of watching and injected SCSS and JS changes?

**Comment by ogdenstudios** [comment]
Do you have anything in your build process set up to process SCSS? Eleventy won't *quite* do that for you. 

You can use [passthrough file copy](https://www.11ty.dev/docs/copy/) or [concatenation and including](https://www.11ty.dev/docs/quicktips/concatenate/) to pass CSS into your layouts or templates.

But the line: 

```
templateFormats: ['njk', 'md', 'scss', 'css', 'html', 'yml'],
```

May be what's tripping you up. CSS and SCSS are not technically `templateFormats`

**Comment by arrowtype** [comment]
@ogdenstudios ahhh I think this clarifies a lot. So, am I correct in interpreting that...

> Tell Eleventy to watch the output files from the build step

...means that the `addWatchTarget()` function is there purely to watch files in `dist`/`_site` and inject those into my open web browser? It is _not_ meant to cue any kinds of file processing, and this should be handled by other tools?

**Comment by ogdenstudios** [comment]
Not quite! 

`addWatchTarget()` will trigger an *Eleventy build*, re: [the documented updates in 0.10](https://www.11ty.dev/docs/config/)

But an Eleventy build *can't process SCSS on its own*. 

You need to configure some other package to process SCSS into CSS. 

[Phil Hawksworth has an article about using SASS with Eleventy that might also help clarify things](https://www.hawksworx.com/blog/keeping-sass-simple-and-speedy-on-eleventy/)

**Comment by arrowtype** [comment]
Right, I think we're saying the same thing, but I didn't say so as clearly. 

Thanks for all your help, @ogdenstudios!

That's a really nice article from Phil Hawksworth.

For now, I've ditched my earlier test, and I'm just working from https://github.com/ianrose/deventy, because I also need JS compilation, and I have no preference between browserSync and liteserver. However, I may be converting a jekyll project soon, so this is great to better understand the scope of Eleventy.

**Comment by arrowtype** [comment]
I've closed this thread, but I do feel like it might be worth saying something in the docs to point out the _purpose_ of `addWatchTarget`. Hawksworth's example of updating inlined CSS seems like one good usecase. :)

**Comment by svivian** [comment]
I know this is an old issue but for in case it's useful, what I do is run Sass (via gulp) and Eleventy simultaneously. Like this:

    (gulp watch & eleventy --quiet --serve)

The parentheses are important. I set `addPassthroughCopy` on my css directory. My SCSS files are in the `_assets` directory, and compile to the css directory, then Eleventy copies those files to the site.

**Comment by cmcknight** [comment]
I use the following .eleventy.js file to compile the SASS files to the _site folder (build in my case):

```
const currentEnv = process.env.ELEVENTY_ENV;

module.exports = function (eleventyConfig) {
  eleventyConfig.addPassthroughCopy('src/js');
  eleventyConfig.addPassthroughCopy('src/css');
  eleventyConfig.addPassthroughCopy('src/data');
  eleventyConfig.addPassthroughCopy('src/img');

  eleventyConfig.addWatchTarget('./src/scss');

  return {
    dir: {
      output: 'build',
      input: 'src',
      data: 'data',
      includes: 'partials_layouts',
    },
    templateFormats: ['njk', 'md', 'html', 'yml'],
    htmlTemplateEngine: "njk"
  };
};
```

The addWatchTarget() triggers a build when an SCSS file changes which refreshes the browser. My package.json file has the following:

```

  "scripts": {
    "dev": "ELEVENTY_ENV=dev npx @11ty/eleventy --serve",
    "sass": "sass src/scss:build/css",
    "watch:sass": "npm run sass -- --watch",
    "watch:eleventy": "npx @11ty/eleventy --serve",
    "start": "npm-run-all sass --parallel watch:*"
  },
```

and

```
"dependencies": {
    "@11ty/eleventy": "^0.12.1",
    "debug": "^4.3.2",
    "node-sass": "^6.0.0",
    "npm-run-all": "^4.1.5"
  }
```

**Comment by patricknelson** [comment]
Great advice above from @ogdenstudios and @svivian. **Bottom line** for now: Don't use 11ty to try to _only_ watch/rebuild SCSS files _in addition_ to it's existing full rebuild capability. At least, not yet.

Sadly, I got myself all tangled up into knots before I ended up finding this thread and ultimately conceding that my original path to accomplishing this was deeply flawed. Being new to 11ty (and seeing soo many Sass plugins), I figured that _**maybe**_ there was some kind of way to setup a plugin that could detect that 11ty was currently in watch mode without some kind of hacky override of the core `Eleventy` class methods (an anti-pattern I've identified in a few plugins).

11ty's awesome, but the documentation is _slightly_ misleading. After first, I experimented with `eleventy-plugin-dart-sass` which triggers a full rebuild. Then, I found `eleventy-plugin-sass` which neatly only compiles `*.scss` files, but its using the holder `node-sass` library and, worse, once upgrading it I found it had some pretty nasty race conditions only visible on build in Netlify due to it's hacky overrides (noted above). 

Overall the developer experience is great, as long as you don't end up accidentally mistaking that 11ty is capable of something that it is not (like I did, d'oh!).
